cli/cli.js

```diff
@@ -1,24 +1,39 @@
 #!/usr/bin/env node
-const { spawn } = require('child_process');
+const { spawn, spawnSync } = require('child_process');
 const path = require('path');
 const fs = require('fs');
 
 let electronBinary = process.env.ELECTRON_BINARY;
 
 if (!electronBinary) {
-  let pathStr =
-    process.platform === 'win32'
-      ? '../node_modules/electron/dist/electron.exe'
-      : '../node_modules/.bin/electron';
-
-  // recursively look for electron binary in node_modules folder
-  for (let i = 0; i < 7; i++) {
-    const maybeElectronBinary = path.resolve(__dirname, pathStr);
-    if (fs.existsSync(maybeElectronBinary)) {
-      electronBinary = maybeElectronBinary;
-      break;
+  // Check whether electron is installed globally and compare with the expected version
+  const electronHandleTemp = spawnSync('electron', ['--version']);
+
+  if (electronHandleTemp.stdout) {
+    if (!electronHandleTemp.stdout.toString().startsWith('v29.')) {
+      console.warn(
+        'WARNING: Found a globally installed electron version but it does not match the version requirements of hc-spin (v29.x). The electron binary from node_modules will be used instead.',
+      );
     } else {
-      pathStr = '../' + pathStr;
+      electronBinary = 'electron';
+    }
+  }
+
+  if (!electronBinary) {
+    let pathStr =
+      process.platform === 'win32'
+        ? '../node_modules/electron/dist/electron.exe'
+        : '../node_modules/.bin/electron';
+
+    // recursively look for electron binary in node_modules folder
+    for (let i = 0; i < 7; i++) {
```

Comment:
is there any significance to this number?

Reply:
No :) It just needs to be high enough for it to work in all sane cases. I'm not particularly proud of the logic but it works.